js/data/buffer/circle_vertex_buffer.js

```diff
@@ -1,41 +1,31 @@
 'use strict';
 
 var util = require('../../util/util');
-var Buffer = require('./buffer');
-
-module.exports = CircleVertexBuffer;
-
-/**
- * This contains the data that displays circle markers on the map,
- * including their centerpoint
- */
-function CircleVertexBuffer(buffer) {
-    Buffer.call(this, buffer);
+var Buffer2 = require('../buffer2');
+
+function CircleVertexBuffer(options) {
+    Buffer2.call(this, options || {
+        type: Buffer2.BufferType.ELEMENT,
```

Comment:
The class name indicates what type of buffer it is. `CircleVertexBuffer` is a vertex buffer. Every call to `draw` in GL requires both an element buffer and a vertex buffer. All shapes are fundamentally drawn in terms of triangles, even lines and circles.